homeassistant/components/vesync/sensor.py

```diff
@@ -144,6 +149,61 @@ def state_class(self):
         """Return the total_increasing state class."""
         return SensorStateClass.TOTAL_INCREASING
 
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes of the device."""
+        if not hasattr(self.smartplug, "weekly_energy_total"):
+            return {}
+        return {
+            "weekly_energy_total": self.smartplug.weekly_energy_total,
```

Comment:
These should also be separate sensors, one sensor per measurement.

Reply:
We've typed the device parameter incorrectly in the existing code. It should not be typed as `VeSyncDevice`. That is actually an entity class, while the `device` parameter represents an instance from the pyvesync library.

A separate PR that fixes the typing is welcome.

Reply:
Fair enough - I can try that. Do you have a pointer to an example component that could serve as a model to follow?

Reply:
Device type fix here: https://github.com/home-assistant/core/pull/73034